essential-projects/metadata
# implement global metadata storage variable

Closes https://github.com/essential-projects/metadata/issues/7

## What did you change?

The `MetadataProvider` now uses the global variable to store the metadata. As soon this changes are published in `essential-projects/metadata@2.0.0` and there is no version 1 around, multiple instances of the metadata provider should no longer result in partial data fragmentation between the instances. 

## How can others test the changes?

To verify that no decorator data is lost due to multiple instances of the `metadata` package, I linked it locally with this feature branch.

In my local setup, I duplicated the metadata package locally so that there are two folders present:
```
./process_engine_meta
   - _integration_tests
   - metadata
   - metadata.1
```

I then ran `npm install` which takes the package under the path `./process_engine_meta/metadata` and uses that module wherever a dependency on `@essential-projects/metadata` exists.

To produce multiple instances I then ran `npm link` in the copied module under the path `./process_engine_meta/metadata.1`.

Finally, to use the copied module as a dependency inside the stack, I navigated to the path: `./process_engine_meta/_integration_tests/node_modules/@process-engine/process_engine/node_modules`. There I removed the symlink on the metadata package created during `npm install` and ran `npm link @essential-projects/metadata` to create a symlink to the duplicated package.

Before running `npm test` you should reset your database. The changes described above specifically affected the `@process-engine/process_engine` package.

If the multiple instances of the `@essential-projects/metadata` caused an error, this should lead to the process engine database tables not being created during migration, because their schemas are missing.

If all tables have been created and the tests have successfully run, the fix should have worked and we no longer need the warning message.

## PR-Checklist

Please check the boxes in this list after submitting your PR:
